### DISCUSSION TITLE
Cannot build under OpenSuse

### DISCUSSION THREAD

**Discussion by juricb71** [moneymanagerex/moneymanagerex#4082] [General]
After having cloned mmex source from branch master, as described in BUILD.d, I'm trying to build it on OpenSuse 15.2.

I built **wxWidgets 3.1.5**:

```
../configure --disable-shared --enable-webview --enable-cxx11 --with-cxx=11 --without-libtiff --enable-universal-binary=arm64,x86_64
make
```
Then I tried to build mmex, created a `build` directory, then:

```
cmake -DCMAKE_CXX_FLAGS="-w" \
-DwxWidgets_CONFIG_EXECUTABLE=/home/juri/Software/wxWidgets/wxWidgets-3.1.5/build-cocoa/wx-config \
-DCMAKE_BUILD_TYPE=Release \
-DCMAKE_OSX_ARCHITECTURES="arm64;x86_64" ..
```
and finally:

`cmake --build . --target package`

When it's finally time for c++ compiling, I can't see why I run into several symbols not declared.

```
Building CXX object src/CMakeFiles/mmex.dir/general_report_manager.cpp.o
In file included from moneymanagerex/src/general_report_manager.cpp:21:0:
moneymanagerex/src/general_report_manager.h:105:22: error: ‘wxWebViewEvent’ has not been declared
     void OnNewWindow(wxWebViewEvent& evt);
                      ^~~~~~~~~~~~~~
moneymanagerex/src/general_report_manager.h:110:5: error: ‘wxWebView’ does not name a type; did you mean ‘wxListView’?
     wxWebView* browser_;
     ^~~~~~~~~
     wxListView
```
and many others follows.

It seems to me that symbol `wxUSE_WEBVIEW`, needed to actually include declarations from file `wx/webview.h` is not defined when file `general_report_manager.h` is included from `general_report_manager.cpp`.

The symbol `wxUSE_WEBVIEW` is defined in `wxWidgets/wxWidgets-3.1.5/include/wx/wx/setup.h` and it's indirectly included into `general_report_manager.cpp`, but not `before general_report_manager.h`.

I've already referred to [https://github.com/moneymanagerex/moneymanagerex/issues/3853](https://github.com/moneymanagerex/moneymanagerex/issues/3853) but the only improvement was to add `--enable-webview` when configuring **wxWidgets** for build.

Any help is appreciated.

**Comment by PMaff** [comment]
Hello, only have seen this now.
Can you check?
:
https://github.com/moneymanagerex/moneymanagerex/issues/4154#issuecomment-1104336216

In short: you have to pay attention what the configure of wxWidget says.
If libwebkit2gtk development stuff is missing, configure of wxWidget silently switches off wxWebView even if you enable it.
